app/services/transaction.service.js

```diff
@@ -21,12 +21,16 @@ const connectorRefundFailureReasons = {
   AMOUNT_BELOW_MINIMUM: 'amount_min_validation'
 }
 
-const searchLedger = async function searchLedger (gatewayAccountIds = [], filters) {
+const searchLedger = async function searchLedger (gatewayAccountIds = [], filters, allServices = false) {
   try {
     return await Ledger.transactions(gatewayAccountIds, filters)
   } catch (error) {
     if (error.errorCode === 504) {
-      throw new GatewayTimeoutError('Your request has timed out. Please apply more filters and try again.')
+      if (allServices === true) {
```

Comment:
could be simplified to `if (allServices) {`

Reply:
👍